validators/tigerDelta/map.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+var linematch = require('linematch');
+var lineclip = require('lineclip');
+
+module.exports = function(data, tile, writeData, done) {
+  var osmlint = 'tigerdelta';
+
+  //filter and normalize input geometry
+  var tiger = toLines(data.tiger.tiger2015);
+  var streets = toLines(data.osm.osm);
+
+  //find tiger parts that are not covered by streets within 10 pixels;
+  //filter out chunks that are too short
+  var diff = linematch(tiger, streets, 20).filter(filterShort);
+  if (diff.length) {
+    var fc = {
+      'type': 'FeatureCollection',
+      'features': []
+    };
+
+    toGeoJSON(diff, tile).forEach(function(line) {
+      var feature = {
+        type: 'Feature',
+        properties: {
+          _osmlint: osmlint
+        },
+        geometry: {
+          type: 'LineString',
+          coordinates: line
+        }
+      };
+      fc.features.push(feature);
+    });
+    writeData(JSON.stringify(fc) + '\n');
+  }
+
+  done(null, null);
+};
+
+function toGeoJSON(diff, tile) {
+  var size = 4096 * Math.pow(2, tile[2]);
+  var x0 = 4096 * tile[0];
+  var y0 = 4096 * tile[1];
+
+  for (var i = 0; i < diff.length; i++) {
+    for (var j = 0; j < diff[i].length; j++) {
+      var p = diff[i][j];
+      var y2 = 180 - (p[1] + y0) * 360 / size;
+      diff[i][j] = [
+        round((p[0] + x0) * 360 / size - 180),
+        round(360 / Math.PI * Math.atan(Math.exp(y2 * Math.PI / 180)) - 90)
+      ];
+    }
+  }
+  return diff;
+}
+
+function round(num) {
+  return Math.round(num * 1e6) / 1e6;
+}
+
+function toLines(layer) {
+  var lines = [];
+  var bbox = [0, 0, 4096, 4096];
+
+  for (var i = 0; i < layer.length; i++) {
+    var feature = layer.feature(i);
+    //only consider polygon features with Tiger name or OSM highway tag
+    if (feature.type === 2 && (feature.properties.FULLNAME !== '' || feature.properties.highway)) {
+      var geom = feature.loadGeometry();
+      for (var k = 0; k < geom.length; k++) {
+        lineclip(normalizeLine(geom[k], layer.extent), bbox, lines); // clip to tile bbox and add to result
+      }
+    }
+  }
+  return lines;
+}
+
+function normalizeLine(line, extent) {
+  var newLine = [];
+  for (var i = 0; i < line.length; i++) {
+    newLine.push([
+      line[i].x * 4096 / extent,
+      line[i].y * 4096 / extent
+    ]);
+  }
+  return newLine;
+}
+
+function filterShort(line) {
+  return dist(line) >= 30; //line length is at least 30 pixels
```

Comment:
@Rub21 interesting. I'll tag @mourner here if he can add thoughts, but we can merge this.